src/compiler/checker.ts

```diff
@@ -727,6 +727,30 @@ module ts {
             return links.target;
         }
 
+        function resolveAliasAsValue(symbol: Symbol): Symbol {
+            Debug.assert((symbol.flags & SymbolFlags.Alias) !== 0, "Should only get Alias here.");
+            let links = getSymbolLinks(symbol);
+            if (!links.aliasValueTarget) {
+                links.aliasValueTarget = resolvingSymbol;
+                let node = getDeclarationOfAliasSymbol(symbol);
+                let target = resolveAlias(symbol);
+                if ((target.flags & SymbolFlags.Alias) && !(target.flags & SymbolFlags.Value)) {
+                    target = resolveAliasAsValue(target);
+                }
+
+                if (links.aliasValueTarget === resolvingSymbol) {
+                    links.aliasValueTarget = target || unknownSymbol;
+                }
+                else {
+                    error(node, Diagnostics.Circular_definition_of_import_alias_0, symbolToString(symbol));

```

Comment:
interesting.  why report the error here?  instead of in the below else-if block where we hit the recursion?